server/routes/driftRadiator.ts

```diff
@@ -2,34 +2,102 @@ import { type RequestHandler, Router } from 'express'
   const post = (path: string, handler: RequestHandler) => router.post(path, asyncMiddleware(handler))
 
   get('/', async (req, res) => {
+    if (req.query.updateServiceArea === '' && isValidDropDown(req, 'serviceArea')) {
+      return res.redirect(`/drift-radiator/service-areas/${req.query.serviceArea}`)
+    }
+    if (req.query.updateTeam === '' && isValidDropDown(req, 'team')) {
+      return res.redirect(`/drift-radiator/teams/${req.query.team}`)
+    }
+    if (req.query.updateProduct === '' && isValidDropDown(req, 'product')) {
+      return res.redirect(`/drift-radiator/products/${req.query.product}`)
+    }
+
     const components = await componentNameService.getAllDeployedComponents()
-    return res.render('pages/driftRadiator', { title: 'Deployment Drift Radiator', components })
+    const serviceAreaList = await dataFilterService.getServiceAreasDropDownList({
+      serviceAreaName: '',
+      useFormattedName: true,
+    })
+    const teamList = await dataFilterService.getTeamsDropDownList({ teamName: '', useFormattedName: true })
+    const productList = await dataFilterService.getProductsDropDownList({ productName: '', useFormattedName: true })
+
+    return res.render('pages/driftRadiator', {
+      title: 'Deployment Drift Radiator',
+      components,
+      serviceAreaList,
+      teamList,
+      productList,
+    })
   })
 
   get('/teams/:teamName', async (req, res) => {
     const { teamName } = req.params
     const components = await componentNameService.getAllDeployedComponentsForTeam(teamName)
-    return res.render('pages/driftRadiator', { title: `Deployment drift radiator for ${teamName}`, components })
+    const serviceAreaList = await dataFilterService.getServiceAreasDropDownList({
+      serviceAreaName: '',
+      useFormattedName: true,
+    })
+    const teamList = await dataFilterService.getTeamsDropDownList({ teamName, useFormattedName: true })
+    const productList = await dataFilterService.getProductsDropDownList({ productName: '', useFormattedName: true })
+
+    return res.render('pages/driftRadiator', {
+      title: `Deployment drift radiator for ${teamName}`,
+      components,
+      serviceAreaList,
+      teamList,
+      productList,
+    })
   })
 
   get('/service-areas/:serviceAreaName', async (req, res) => {
     const { serviceAreaName } = req.params
     const components = await componentNameService.getAllDeployedComponentsForServiceArea(serviceAreaName)
-    return res.render('pages/driftRadiator', { title: `Deployment drift radiator for ${serviceAreaName}`, components })
+    const serviceAreaList = await dataFilterService.getServiceAreasDropDownList({
+      serviceAreaName,
+      useFormattedName: true,
+    })
+    const teamList = await dataFilterService.getTeamsDropDownList({ teamName: '', useFormattedName: true })
+    const productList = await dataFilterService.getProductsDropDownList({ productName: '', useFormattedName: true })
```

Comment:
guess these three get called all at the sametime - could add a `getDropDownLists` to the service and reduce service calls here 🤷